java/ql/lib/semmle/code/java/StringCheck.qll

```diff
@@ -0,0 +1,14 @@
+/**
+ * Provides classes and predicates for reasoning about checking characterizations about strings.
+ */
+
+import java
+
+/**
+ * Holds if `ma` is a call to a method that checks a partial string match.
+ */
+predicate isStringPartialMatch(MethodAccess ma) {
+    ma.getMethod().getDeclaringType() instanceof TypeString and
+    ma.getMethod().getName() =
+      ["contains", "startsWith", "matches", "regionMatches", "indexOf", "lastIndexOf"]
```

Comment:
Should this also consider `endsWith`? Otherwise the predicate name and documentation might be a bit misleading.

Reply:
I would be inclined to agree given the method name, but I think that `endsWith` isn't something that would work with either of the current uses of this utility method.

@smowton what do you think? Change the name? I was also thinking about a query being more flexible to include/exclude end checks.

Reply:
If you need a rather irregular predicate like this make it `private`, name it something like `isOsCheckMatchingMethod` and note in the qldoc why it only considers particular methods